src/lib/comms/sol-coap.c

```diff
@@ -682,13 +690,74 @@ timeout_expired(struct sol_coap_server *server, struct outgoing *outgoing)
     return expired;
 }
 
+static struct resource_context *
+find_context(struct sol_coap_server *server,
+    const struct sol_coap_resource *resource)
+{
+    struct resource_context *c;
+    uint16_t i;
+
+    SOL_VECTOR_FOREACH_IDX (&server->contexts, c, i) {
+        if (c->resource == resource)
+            return c;
+    }
+    return NULL;
+}
+
+static int
+notification_pkt_iterate(struct sol_coap_server *server,
+    struct outgoing *outgoing)
+{
+    int r, i;
+    struct resource_observer *o;
+
+    SOL_PTR_VECTOR_FOREACH_IDX (&outgoing->context->observers, o, i) {
+        r = sol_ptr_vector_find_first(&outgoing->notified, o);
+        if (r < 0) {
+            uint8_t *token, tkl;
+
+            r = sol_ptr_vector_append(&outgoing->notified, o);
+            SOL_INT_CHECK(r, < 0, r);
+
+            token = sol_coap_header_get_token(outgoing->pkt, &tkl);
+            if (tkl > outgoing->tkl) {
+                uint8_t bytes[tkl - o->tkl];
```

Comment:
Uninitialised memory? 

Does the compiler in semaphore support this "dynamic" array (I forgot the proper name)? I remember this being relatively "new".